### PR TITLE
fix(bundler): default hashing method doesnt work in node 18

### PR DESCRIPTION
**Pull request by 10xLaCroixDrinker**
#### _Describe your changes in detail_

Addresses: https://github.com/webpack/webpack/issues/14532

#### _Why is this change required? What issue does it resolve?_

Current hashing method is unsupported by Node 17+

#### _How does your change affect the rest of the code._ 

Installed and built in one-app using Node 18

#### Check boxes that apply:
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [x] Dependency update
- [ ] Security update

#### Check boxes that apply:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] These changes should be applied to a maintenance branch.
- [x] I have added the Apache 2.0 license header to any new files created.

### CODE REVIEW

File: packages/one-app-bundler/webpack/webpack.common.js [hunk 1]
```diff
@@ -41,6 +41,9 @@ const plugins = [
 module.exports = {
   profile: true,
   devtool: nodeEnvironmentIsProduction ? false : 'source-map',
+  output: {
+    hashFunction: 'xxhash64',
```
Comment (10xLaCroixDrinker) [inline packages/one-app-bundler/webpack/webpack.common.js:45]: The default hashing method is not available in Node 18

File: packages/generator-one-app-module/generators/app/index.js [hunk 2]
```diff
@@ -13,12 +13,17 @@
...
 
 /* eslint-disable no-underscore-dangle -- yeoman private methods use underscores */
+const path = require('path');
 const Generator = require('yeoman-generator');
 const chalk = require('chalk');
 const _ = require('lodash');
 const helper = require('./promptValidations');
 const packagejs = require('../../package.json');
 
+_.extend(Generator.prototype, require('yeoman-generator/lib/actions/install'));
```
Comment (10xLaCroixDrinker) [inline packages/generator-one-app-module/generators/app/index.js:23]: Yeoman 5.0 removed the install hook by default. This restores it.

File: packages/generator-one-app-module/generators/app/index.js [hunk 3]
```diff
@@ -13,12 +13,17 @@
...
+
+const getTemplatePath = (templateName) => path.join(__dirname, 'templates', templateName);
```
Comment (10xLaCroixDrinker) [inline packages/generator-one-app-module/generators/app/index.js:25]: Yeoman recommends absolute paths for more deterministic results. Tests also failed without this change after updating to Yeoman 5

File: packages/generator-one-app-module/package.json [hunk 4]
```diff
@@ -30,12 +30,11 @@
   "dependencies": {
     "chalk": "^4.1.0",
     "lodash": "^4.17.19",
-    "yeoman-generator": "^4.11.0"
+    "yeoman-environment": "^3.10.0",
```
Comment (10xLaCroixDrinker) [inline packages/generator-one-app-module/package.json:33]: New peerDep required in Yeoman 5

**Review by JAdshead — APPROVED** [review]

**Review by bishnubista — APPROVED** [review]

**Pull request merged**